cluster-operator/src/main/java/io/strimzi/operator/cluster/model/EntityOperator.java

```diff
@@ -319,4 +322,38 @@ public Role generateRole(String ownerNamespace, String namespace) {
 
         return role;
     }
+
+    /**
+     * Generates the NetworkPolicies relevant for Entity Operator
+     *
+     * @return The network policy.
+     */
+    public NetworkPolicy generateNetworkPolicy() {
+        // List of network policy rules for all ports
+        List<NetworkPolicyIngressRule> rules = new ArrayList<>();
+
+        // For Topic Operator
+        if (topicOperator != null) {
+            rules.add(NetworkPolicyUtils.createIngressRule(EntityTopicOperator.HEALTHCHECK_PORT, List.of()));
+        }
+
+        // For User Operator
+        if (userOperator != null) {
+            rules.add(NetworkPolicyUtils.createIngressRule(EntityUserOperator.HEALTHCHECK_PORT, List.of()));
+        }
+
+
+        // Everyone can access metrics
+        rules.add(NetworkPolicyUtils.createIngressRule(MetricsModel.METRICS_PORT, List.of()));
+
+
+        // Build the final network policy with all rules covering all the ports
+        return NetworkPolicyUtils.createNetworkPolicy(
+                reconciliation.name() + "-network-policy-entity-operator",
```

Comment:
I would just use `componentName` here.

```suggestion
                componentName,
```